src/game/magic/spells/SpellsLvl01.cpp

```diff
@@ -56,25 +56,35 @@ void MagicSightSpell::Launch() {
 	m_hasDuration = m_launchDuration >= 0;
 	m_duration = m_hasDuration ? m_launchDuration : 0;
 	
-	ARX_SOUND_PlaySFX(g_snd.SPELL_VISION_START, &m_caster_pos);
+	bool emitsSound = !(m_flags & SPELLCAST_FLAG_NOSOUND);
```

Comment:
this bit flag magic could be moved to a macro or function instead. makes it easier to understand


    bool emitsSound = GET_EMITS_SOUND(m_flags);

also removes code repetition

Reply:
Why use a macro when a member function could look much cleaner? I lean towards keeping the flag checks as they are, they are fairly clear.

Reply:
there is a double negation in there (NO and !) so they are much harder to read than "getEmitsSound"

a macro would mean no extra call to a function for such a simple operation

Reply:
As far as I'm concerned, macros are just preprocessor crutches. I wouldn't worry about efficiency as the function would probably get inlined, and this is not a hot codepath anyway.

Reply:
I would also lean towards a method in the Spell class as opposed to adding a macro because macros are leaning the codebase back towards C and low level constructs while a method could be migrated more easily to a more modern language if needed.